src/Http/ServerRequest.php

```diff
@@ -577,16 +584,54 @@ public function session(Session $session = null)
      */
     public function clientIp()
     {
-        if ($this->trustProxy && $this->getEnv('HTTP_X_FORWARDED_FOR')) {
-            $addresses = explode(',', $this->getEnv('HTTP_X_FORWARDED_FOR'));
-            $ipaddr = end($addresses);
-        } elseif ($this->trustProxy && $this->getEnv('HTTP_CLIENT_IP')) {
-            $ipaddr = $this->getEnv('HTTP_CLIENT_IP');
-        } else {
-            $ipaddr = $this->getEnv('REMOTE_ADDR');
+        if ($this->trustProxy) {
+            if ($forwarded = $this->getEnv('HTTP_X_FORWARDED_FOR')) {
```

Comment:
We avoid assignment in conditionals as it can be harder to read and it has a number of edge cases.